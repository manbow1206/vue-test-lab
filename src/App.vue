<script setup>
import Hello from './components/HelloWorld.vue';
import { ref } from 'vue';

const name = ref('John');

const handleEvent = (newName) => {
  name.value = `${newName.firstName} ${newName.lastName}`;
};
</script>

<template>
  <h1>Vue 3 入門</h1>
  <Hello @changeNameEvent="handleEvent" :name="name" />
</template>

<style>
#app {
  font-family: Avenir, Helvetica, Arial, sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  text-align: center;
  color: #2c3e50;
  margin-top: 60px;
}
</style>
