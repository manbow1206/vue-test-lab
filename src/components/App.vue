<template>
  <p v-if="message">{{ message }}</p>
  <a href="#"></a>

</template>

<script>
export default {
  props: {
    message: {
      type: String,
      required: false,
      default: '',
    },
  },
};
</script>