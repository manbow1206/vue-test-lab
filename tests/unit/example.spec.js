import { shallowMount } from '@vue/test-utils';
import App from '@/components/App.vue';

test('何もレンダリングされないケース', () => {
  const wrapper = shallowMount(App);

  expect(wrapper.element).toMatchSnapshot();
});

it('文字列を渡すケース', () => {
  const wrapper = shallowMount(App, {
    propsData: {
      message: 'Vue Test Utils',
    },
  });

  expect(wrapper.element).toMatchSnapshot();
});
