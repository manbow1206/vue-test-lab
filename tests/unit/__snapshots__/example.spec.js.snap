// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`何もレンダリングされないケース 1`] = `
<div
  data-v-app=""
>
  
  <!--v-if-->
  <a
    href="#"
  />
  
</div>
`;

exports[`文字列を渡すケース 1`] = `
<div
  data-v-app=""
>
  
  <p>
    Vue Test Utils
  </p>
  <a
    href="#"
  />
  
</div>
`;
